test/react/first-render-only/simple-3.js

```diff
@@ -0,0 +1,40 @@
+var React = require('react');
+// the JSX transform converts to React, so we need to add it back in
+this['React'] = React;
+var checkRef = false;
+
+function App(props) {
+  function fn() {
+    // should not be here
+  }
+  function fn2() {
+    return props.bar(fn);
+  }
+  return (
+    <div onClick={fn} ref={fn2} />
+  );
+}
+
+App.getTrials = function(renderer, Root) {
+  let val;
+  function func(_val) {
+    val = _val;
+  }
+  renderer.update(<Root bar={func} />);
+  let results = [];
+  results.push(['simple render', renderer.toJSON()]);
+  if (checkRef === true && val !== undefined) {
```

Comment:
I think this would make the test intention clearer

```js
if (isOptimizedForFirstRender && val !== undefined) {
  throw new Error('Expected first render mode to not serialize ref callbacks.');
}
```

:-)

I initially was a bit puzzled as to what we're testing.

Reply:
Nice one, will do! :)